helpers/TagHelper.js

```diff
@@ -0,0 +1,43 @@
+import db from '../models';
+
+const { Tag } = db;
+
+
+/**
+ * @exports TagHelper
+ * @class TagHelper
+ * @description Contains methods to assist Tag
+ * */
+class TagHelper {
+  /**
+   *
+   * @param {Array<Tag>}tags
+   * @return {Promise<Array>} returns a Promise
+   */
+  static async findOrCreateTags(tags) {
+    const tagList = await Promise.all(TagHelper.getTagObj(tags));
+    const tagData = [];
+    tagList.forEach((tag) => {
+      tagData.push(tag[0]);
+    });
+    return tagData;
+  }
+
+  /**
+   *
+   * @param {Array}tags
+   * @return {Array} Returns array of tag object to be stored in the databse
+
+   */
+  static getTagObj(tags) {
+    const result = [];
+    if (tags) {
+      tags.forEach((tag) => {
+        result.push(Tag.findOrCreate({ where: { name: tag } }));
```

Comment:
Nice work on making the database operations fully asynchronous. I think it will be nice to check if the tag is an empty string too. The current implementation allows empty strings to be stored to the database as tags.

Reply:
Nice catch. Thanks

Reply:
Have you resolved this?